diff --git a/build/modern/chunk-4FWBSZRD.js b/build/modern/chunk-4FWBSZRD.js
index 31694707f095bed7e775fa40fcace64a88a5ca59..b28f99170f383892e422f2ccabffe8ad4e249b22 100644
--- a/build/modern/chunk-4FWBSZRD.js
+++ b/build/modern/chunk-4FWBSZRD.js
@@ -874,7 +874,6 @@ var rule5 = createPropertyOrderRule(
 
 // src/rules/no-void-query-fn/no-void-query-fn.rule.ts
 import { ESLintUtils as ESLintUtils6 } from "@typescript-eslint/utils";
-import ts from "typescript";
 var name6 = "no-void-query-fn";
 var createRule6 = ESLintUtils6.RuleCreator(getDocsUrl);
 var rule6 = createRule6({
@@ -926,7 +925,7 @@ function isIllegalReturn(checker, type) {
   if (awaited.isUnion()) {
     return awaited.types.some((t) => isIllegalReturn(checker, t));
   }
-  return awaited.flags & (ts.TypeFlags.Void | ts.TypeFlags.Undefined) ? true : false;
+  return awaited.flags & (16384 | 32768) ? true : false;
 }
 
 // src/rules/mutation-property-order/constants.ts
