diff --git a/build/modern/chunk-WNWA3ZM7.js b/build/modern/chunk-WNWA3ZM7.js
index 63a44419c85b533fff6fee482f239a8c6609d3f6..11a57ae3bda79543eb93ff63af4e14c0dbbb86db 100644
--- a/build/modern/chunk-WNWA3ZM7.js
+++ b/build/modern/chunk-WNWA3ZM7.js
@@ -893,7 +893,6 @@ var rule5 = createPropertyOrderRule(
 
 // src/rules/no-void-query-fn/no-void-query-fn.rule.ts
 import { ESLintUtils as ESLintUtils6 } from "@typescript-eslint/utils";
-import ts from "typescript";
 var name6 = "no-void-query-fn";
 var createRule6 = ESLintUtils6.RuleCreator(getDocsUrl);
 var rule6 = createRule6({
@@ -945,7 +944,7 @@ function isIllegalReturn(checker, type) {
   if (awaited.isUnion()) {
     return awaited.types.some((t) => isIllegalReturn(checker, t));
   }
-  return awaited.flags & (ts.TypeFlags.Void | ts.TypeFlags.Undefined) ? true : false;
+  return awaited.flags & (16384 | 32768) ? true : false;
 }
 
 // src/rules/mutation-property-order/constants.ts